File: py/49.py
from math import sqrt

def prime(n):
    if n % 2 == 0:
        return False

    for i in range(3, int(sqrt(n)) + 1):
        if n != i and n % i == 0:
            return False

    return True

def same_digits(ns):
    return len(set(map(lambda n: ''.join(sorted(list(str(n)))), ns))) == 1

def solution(n):
    ns = [n, n + 3330, n + 6660]
    return all(map(prime, ns)) and same_digits(ns)

sol = next(n for n in xrange(1488, 3000) if solution(n))
print '%d%d%d' % (sol, sol + 3330, sol + 6660)

File: js/49.js
function prime(n) {
    if (n == 2)
        return true;

    if (n % 2 == 0)
        return false;

    for (m = 3; m <= Math.sqrt(n); m += 2)
        if (n % m == 0)
            return false;

    return true;
}

function same_digits(a, b) {
    return a.toString().split('').sort().toString()
        == b.toString().split('').sort().toString();
}

for (var i = 1488 ;; ++i) {
    var j = i + 3330, k = i + 6660;
    if (prime(i) && prime(j) && prime(k) && same_digits(i, j) && same_digits(j, k))
        break;
}

console.log([i, i + 3330, i + 6660].join(''));
